src/protocol/sort/reveal.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    error::BoxError,
+    field::Field,
+    helpers::{
+        mesh::{Gateway, Mesh},
+        prss::SpaceIndex,
+        Direction,
+    },
+    protocol::{context::ProtocolContext, RecordId, Step},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've revealed their own shares
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct AValue<F> {
+    a: F,
+}
+
+/// This implements reveal algorithm of "Efficient Secure Three-Party Sorting Protocol with an Honest Majority" at communication cost of 3R.
+/// Revealing is the protocol that on input a pair of shares, outputs a secret. For simplicity, we
```

Comment:
Why "a pair of shares"? Doesn't it take all 3 secret-shares of just one sharing?

Reply:
it already has 2 shares of its own (which is obtained as input). While running the protocol, it receives the 3rd share and computes the secret. Edited to make it clearer